### PR TITLE
Azure stack deployment resource provider admin open api specification

### PR DESCRIPTION
**Pull request by viananth**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by lirenhe** [comment]
@viananth, please fix those code check errors in Avocado and ModelValidation.

**Comment by viananth** [comment]
> @viananth, please fix those code check errors in Avocado and ModelValidation.

@lirenhe We are having an issue with representing JToken format. Could you please help how we should specify in the spec?

`error : 

operationId: ActionPlanOperations_Get

scenario: Gets the specified action plan operation

source: response

responseCode: '200'

severity: 0

code: INVALID_TYPE

details:

  code: INVALID_TYPE

  params:

    - string

    - object

  message: Expected type string but found type object

  path: properties/outputs

  title: '#/definitions/ActionPlanOperationAdminProperties/properties/outputs'

  description: **The action plan operation outputs in JToken format**

  position:

    line: 162

    column: 20

  url: >-

    specification/azsadmin/resource-manager/deployment/Microsoft.Deployment.Admin/preview/2019-01-01/ActionPlanOperation.json

  directives: {}


  jsonUrl: >-

    specification/azsadmin/resource-manager/deployment/Microsoft.Deployment.Admin/preview/2019-01-01/examples/ActionPlanOperations/Get.json

  jsonPath: $.properties.outputs
`

**Comment by lirenhe** [comment]
> @viananth, please fix those code check errors in Avocado and ModelValidation.
> 
> @lirenhe We are having an issue with representing JToken format. Could you please help how we should specify in the spec?
> error :  operationId: ActionPlanOperations_Get scenario: Gets the specified action plan operation source: response responseCode: '200' severity: 0 code: INVALID_TYPE details: code: INVALID_TYPE params: - string - object message: Expected type string but found type object path: properties/outputs title: '#/definitions/ActionPlanOperationAdminProperties/properties/outputs' description: **The action plan operation outputs in JToken format** position: line: 162 column: 20 url: >- specification/azsadmin/resource-manager/deployment/Microsoft.Deployment.Admin/preview/2019-01-01/ActionPlanOperation.json directives: {} jsonUrl: >- specification/azsadmin/resource-manager/deployment/Microsoft.Deployment.Admin/preview/2019-01-01/examples/ActionPlanOperations/Get.json jsonPath: $.properties.outputs

@viananth, I didn't see that error in Avocado and Modelvalidation errors, is this already fixed?

**Comment by lirenhe** [comment]
@viananth, wrong click to remove your comments. Could you leave this issue for now and fix other problems first?

**Comment by bganapa** [comment]
@lirenhe I posted the question on Teams Channel....
Is there a precedence on how to specify JToken parameter in swagger object model?We have specified the type as string in for the JToken param.  The Oav validate-example fails as the example is not matching to the defined type string while it is expecting object

**Comment by lirenhe** [comment]
> @lirenhe I posted the question on Teams Channel....
> Is there a precedence on how to specify JToken parameter in swagger object model?We have specified the type as string in for the JToken param.  The Oav validate-example fails as the example is not matching to the defined type string while it is expecting object

@bganapa , thanks for sharing the detailed information, we will investigate this issue.

**Comment by viananth** [comment]
@lirenhe I have this avocado failure, where as it fails to resolve $(this-folder) for one json but resolves the same for other json files. I can't repro this on my local dev machine. Seems like an issue in the build. Could you pls take a look?

```
error: 

code: NO_JSON_FILE_FOUND

message: The JSON file is not found but it is referenced from the readme file.

readMeUrl: >-

  /home/vsts/work/1/c93b354fd9c14905bb574a8834c4d69b/specification/azsadmin/resource-manager/deployment/readme.md

level: Error

jsonUrl: >-

  /home/vsts/work/1/c93b354fd9c14905bb574a8834c4d69b/specification/azsadmin/resource-manager/deployment/$(this-folder)/Microsoft.Deployment.Admin/preview/2019-01-01/ProductSecret.json
```

**Comment by viananth** [comment]
I fixed the Model validation errors and it doesn't repro on my dev machine. Not sure why CI is still complaining about it:

```
 error : 

operationId: FileContainers_Create

scenario: Creates a new product package.

source: response

responseCode: '200'

severity: 0

code: OBJECT_ADDITIONAL_PROPERTIES

details:

  code: OBJECT_ADDITIONAL_PROPERTIES

  params:

    - 'properties '

  message: 'Additional properties not allowed: properties '


  path: ''

  title: '#/definitions/FileContainer'

  description: File container entity.

  position:

    line: 163

    column: 26

  url: >-

    specification/azsadmin/resource-manager/deployment/Microsoft.Deployment.Admin/preview/2019-01-01/FileContainer.json

  directives: {}

  jsonPosition:

    line: 16

    column: 15

  jsonUrl: >-

    specification/azsadmin/resource-manager/deployment/Microsoft.Deployment.Admin/preview/2019-01-01/examples/FileContainer/Create.json

  jsonPath: ''
```

File: specification/azsadmin/resource-manager/deployment/Microsoft.Deployment.Admin/preview/2019-01-01/examples/FileContainer/Create.json
Comment (lirenhe) [inline specification/azsadmin/resource-manager/deployment/Microsoft.Deployment.Admin/preview/2019-01-01/examples/FileContainer/Create.json]: Please remove the additional space in the property name and value

File: specification/azsadmin/resource-manager/deployment/Microsoft.Deployment.Admin/preview/2019-01-01/ProductPackage.json
Comment (lirenhe) [inline specification/azsadmin/resource-manager/deployment/Microsoft.Deployment.Admin/preview/2019-01-01/ProductPackage.json]: Is this the correct name of the example?

File: specification/azsadmin/resource-manager/deployment/Microsoft.Deployment.Admin/preview/2019-01-01/FileContainer.json
Comment (lirenhe) [inline specification/azsadmin/resource-manager/deployment/Microsoft.Deployment.Admin/preview/2019-01-01/FileContainer.json]: Please add the default response

**Comment by lirenhe** [comment]
> I fixed the Model validation errors and it doesn't repro on my dev machine. Not sure why CI is still complaining about it:
>  error : 
> 
> operationId: FileContainers_Create
> 
> scenario: Creates a new product package.
> 
> source: response
> 
> responseCode: '200'
> 
> severity: 0
> 
> code: OBJECT_ADDITIONAL_PROPERTIES
> 
> details:
> 
>   code: OBJECT_ADDITIONAL_PROPERTIES
> 
>   params:
> 
>     - 'properties '
> 
>   message: 'Additional properties not allowed: properties '
> 
> 
>   path: ''
> 
>   title: '#/definitions/FileContainer'
> 
>   description: File container entity.
> 
>   position:
> 
>     line: 163
> 
>     column: 26
> 
>   url: >-
> 
>     specification/azsadmin/resource-manager/deployment/Microsoft.Deployment.Admin/preview/2019-01-01/FileContainer.json
> 
>   directives: {}
> 
>   jsonPosition:
> 
>     line: 16
> 
>     column: 15
> 
>   jsonUrl: >-
> 
>     specification/azsadmin/resource-manager/deployment/Microsoft.Deployment.Admin/preview/2019-01-01/examples/FileContainer/Create.json
> 
>   jsonPath: ''

The model validation failed because you added additional space in the property name and value

**Comment by ruowan** [comment]
For avocado error, currently avocado don't support `${ths-folder}` feature. So you can ignore avocado validation errors which contains '${this-folder}' path.

File: specification/azsadmin/resource-manager/deployment/Microsoft.Deployment.Admin/preview/2019-01-01/ProductPackage.json
Comment (viananth) [reply specification/azsadmin/resource-manager/deployment/Microsoft.Deployment.Admin/preview/2019-01-01/ProductPackage.json]: It looks correct to me. Can you pls elaborate?

File: specification/azsadmin/resource-manager/deployment/Microsoft.Deployment.Admin/preview/2019-01-01/FileContainer.json
Comment (lirenhe) [inline specification/azsadmin/resource-manager/deployment/Microsoft.Deployment.Admin/preview/2019-01-01/FileContainer.json]: Is this the correct name of the example?

File: specification/azsadmin/resource-manager/deployment/Microsoft.Deployment.Admin/preview/2019-01-01/ProductPackage.json
Comment (lirenhe) [reply specification/azsadmin/resource-manager/deployment/Microsoft.Deployment.Admin/preview/2019-01-01/ProductPackage.json]: Put it in the wrong place. Resolve this

**Comment by lirenhe** [comment]
It would be good if you could also fix the code style issue reported by 'PrettierCheck'

**Comment by viananth** [comment]
> It would be good if you could also fix the code style issue reported by 'PrettierCheck'

@lirenhe I fixed the prettier check issues, still the CI is reporting the same. I fixed all the examples as well.

```
D:\github\azure-rest-api-specs>prettier -c ./specification/azsadmin/resource-manager/deployment/Microsoft.Deployment.Admin/preview/2019-01-01/*.*

Checking formatting...
All matched files use Prettier code style!
```

File: specification/azsadmin/resource-manager/deployment/Microsoft.Deployment.Admin/preview/2019-01-01/ActionPlan.json
Comment (lirenhe) [inline specification/azsadmin/resource-manager/deployment/Microsoft.Deployment.Admin/preview/2019-01-01/ActionPlan.json]: I think you need to add 'default' in each response, you could refer this as an example: https://github.com/Azure/azure-rest-api-specs/blob/master/specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/DatabaseSchema.json

Comment (viananth) [reply specification/azsadmin/resource-manager/deployment/Microsoft.Deployment.Admin/preview/2019-01-01/ActionPlan.json]: Prettier Check passed now. It was failing because I had to rebase from master.
@lirenhe Everything looks good now except the known avocado issue. We don't need the "default" value for each response. Could you please merge if it looks good otherwise? Thanks!

**Review by lirenhe — APPROVED** [review]

**Comment by lirenhe** [comment]
@viananth, please wait for ARM team to signoff this PR then I could merge it.

**Comment by bganapa** [comment]
@lirenhe This is for AZURESTACK Admin API surface. This does not get deployed to azure. We have our internal AZURESTACK review process and it has already gone through that process. 

These specs never go through ARM Team sign off. You can look at any PRs for changes under https://github.com/Azure/azure-rest-api-specs/tree/master/specification/azsadmin/resource-manager

**Review by bganapa — APPROVED** [review]

**Comment by lirenhe** [comment]
> @lirenhe This is for AZURESTACK Admin API surface. This does not get deployed to azure. We have our internal AZURESTACK review process and it has already gone through that process.
> These specs never go through ARM Team sign off. You can look at any PRs for changes under https://github.com/Azure/azure-rest-api-specs/tree/master/specification/azsadmin/resource-manager

I checked your last PR https://github.com/Azure/azure-rest-api-specs/pull/7591, it still requires ARM sign off.  As you have this PR reviewed in your internal process, I will merge it.

**Comment by bganapa** [comment]
@lirenhe Thanks for pointing out the last PR. I am pinging the PR owner offline to loop us in.

**Pull request merged**
